src/drivers/cmakeDriver.ts

```diff
@@ -485,78 +485,113 @@ export abstract class CMakeDriver implements vscode.Disposable {
      * Change the current configure preset. This lets the driver reload, if necessary.
      * @param configurePreset The new configure preset
      */
-    async setConfigurePreset(configurePreset: preset.ConfigurePreset): Promise<void> {
-        if (!this.useCMakePresets) {
-            log.info(localize('skip.set.config.preset', 'Using kits, skip setting configure preset: {0}', configurePreset.name));
-            return;
-        }
+    async setConfigurePreset(configurePreset: preset.ConfigurePreset | null): Promise<void> {
```

Comment:
We should prefer APIs that use `undefined` over `null`.  We'd been starting to remove `null` in the codebase and hoped to get all references to it out.  The syntax is cleaner too. `configurePreset?: preset.ConfigurePreset`

Reply:
Agreed. I'll change those touched in this PR.

Reply:
Turns out this requires too many changes. We should change them all together in a separate PR.